src/renderer/lib/cast.js

```diff
@@ -130,22 +132,51 @@ function chromecastPlayer () {
     })
   }
 
+  function serveSubtitles (cb) {
+    const subtitles = state.playing.subtitles
+    const selectedSubtitle = subtitles.tracks[subtitles.selectedIndex]
+    if (!selectedSubtitle) {
+      cb()
+    } else {
+      ret.subServer = http.createServer((req, res) => {
+        if (!selectedSubtitle) {
+          res.writeHead(400, {'Content-Type': 'text/html'})
+          return res.end('No subtitles selected')
+        }
+        res.writeHead(200, {
+          'Content-Type': 'text/vtt;charset=utf-8',
+          'Access-Control-Allow-Origin': '*',
+          'Transfer-Encoding': 'chunked'
+        })
+        res.end(Buffer.from(selectedSubtitle.buffer.substr(21), 'base64'))
+      }).listen(0, function () {
+        const port = ret.subServer.address().port
+        const subtitlesUrl = 'http://' + state.server.networkAddress + ':' + port + '/'
+        cb(subtitlesUrl)
+      })
+    }
+  }
+
   function open () {
     const torrentSummary = state.saved.torrents.find((x) => x.infoHash === state.playing.infoHash)
-    ret.device.play(state.server.networkURL + '/' + state.playing.fileIndex, {
-      type: 'video/mp4',
-      title: config.APP_NAME + ' - ' + torrentSummary.name
-    }, function (err) {
-      if (err) {
-        state.playing.location = 'local'
-        state.errors.push({
-          time: new Date().getTime(),
-          message: 'Could not connect to Chromecast. ' + err.message
-        })
-      } else {
-        state.playing.location = 'chromecast'
-      }
-      update()
+    serveSubtitles(function (subtitles) {
```

Comment:
Since the `subtitles` variable is not actually an array of subtitles it makes more sense to use the same name as used before in the `serveSubtitles` function - namely `subtitlesUrl`.